lib/auth/facebook_authenticator.rb

```diff
@@ -20,7 +63,14 @@ def after_authenticate(auth_token)
 
     user_info = FacebookUserInfo.find_by(facebook_user_id: facebook_hash[:facebook_user_id])
 
-    result.user = user_info.try(:user)
+    if existing_account && (user_info.nil? || existing_account.id != user_info.user_id)
+      user_info.destroy! if user_info
+      result.user = existing_account
+      user_info = FacebookUserInfo.create({ user_id: result.user.id }.merge(facebook_hash))
```

Comment:
it doesn't look like we're checking for the return value of `#create`. Therefore, we'll want to use `create!` here.